File: backend/app/models/module_registry.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


ModuleStatus = Literal["mvp", "planned"]


class OsModule(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str = Field(min_length=1)
    label: str = Field(min_length=1)
    status: ModuleStatus
    api_prefixes: list[str] = Field(min_length=1)
    capabilities: list[str] = Field(min_length=3)
    requires_permission: bool
    audit_events: list[str] = Field(min_length=1)
    rate_limit_policy: str = Field(min_length=1)


class ModuleRegistryResponse(BaseModel):
    modules: list[OsModule]

File: frontend/src/types/moduleRegistry.ts
export type BackendModuleStatus = "mvp" | "planned";
export type ModuleStatusLabel = "MVP" | "规划";

export interface BackendOsModule {
  id: string;
  label: string;
  status: BackendModuleStatus;
  api_prefixes: string[];
  capabilities: string[];
  requires_permission: boolean;
  audit_events: string[];
  rate_limit_policy: string;
}

export interface ModuleRegistryResponse {
  modules: BackendOsModule[];
}

export interface OsModuleItem {
  label: string;
  anchor: string;
}

export interface NavigationOsModule {
  id: string;
  label: string;
  status: ModuleStatusLabel;
  items: OsModuleItem[];
  roadmap: string[];
  nextIntegration: string;
  apiPrefixes?: string[];
  capabilities?: string[];
  requiresPermission?: boolean;
  auditEvents?: string[];
  rateLimitPolicy?: string;
}
